orbit-android/src/main/java/com/babylon/orbit/OrbitViewModel.kt

```diff
@@ -51,16 +56,35 @@ open class OrbitViewModel<STATE : Any, SIDE_EFFECT : Any>(
         stateConsumer: (STATE) -> Unit,
         sideEffectConsumer: (SIDE_EFFECT) -> Unit = {}
     ) {
-        mainThreadOrbit
-            .autoDispose(lifecycleOwner)
-            .subscribe(stateConsumer)
+        disposables += mainThreadOrbit.subscribe(stateConsumer)
+        disposables += mainThreadSideEffect.subscribe(sideEffectConsumer)
 
-        mainThreadSideEffect
-            .autoDispose(lifecycleOwner)
-            .subscribe(sideEffectConsumer)
+        val exitEvent = lifecycleOwner.lifecycle.currentState.correspondingExitEvent
```

Comment:
Can we perhaps extract the lifecycle related logic to a separate class or function with separate, focused unit tests? This feels like too many responsibilities in `OrbitViewModel`. 

Plus, we could then reuse it in our main project (one could argue we could open source it as a micro-library at some point).

The way I see it you should be able to simply go `disposable.autoDispose(lifecycleOwner)` taking inspiration from the original. Feel free to come up with a less rip-off name though 😅 `bindToLifecycle` perhaps?